packages/fmg-nitro-adjudicator/contracts/NitroAdjudicator.sol

```diff
@@ -0,0 +1,420 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+import "fmg-core/contracts/State.sol";
+import "fmg-core/contracts/Rules.sol";
+
+contract NitroAdjudicator {
+    using State for State.StateStruct;
+
+    struct Authorization {
+        // Prevents replay attacks:
+        // It's required that the participant signs the message, meaning only
+        // the participant can authorize a withdrawal.
+        // Moreover, the participant should sign the address that they wish
+        // to send the transaction from, preventing any replay attack.
+        address participant; // the account used to sign state transitions
+        address destination; // either an account or a channel
+        uint amount;
+        address sender; // the account used to sign transactions
+    }
+
+    struct Outcome {
+        address[] destination;
+        uint[] amount;
+        uint256 finalizedAt;
+        State.StateStruct challengeState;
+    }
+    struct Guarantee {
+        address guarantor;
+        address target;
+        address[] priorities;
```

Comment:
Guarantees are equivalent to allocations in the way they get on-chain. From the force-move side it's just another type of outcome. We'll still need finalizedAt etc. on the guarantees.

Reply:
It's possible that guarantees could be stored as outcomes where the amounts are all 0 (and there's an `isGuarantee` flag or similar) - that way all the force-move logic will work unchanged.